### PR TITLE
Add io.github.gssh_linux.gssh

### PR DESCRIPTION
**Pull request by SStoklosa23**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [  ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

# Permissions
As my app is for connecting to ssh servers with a GUI, this are why everything is needed:
I use the network permission for connections
I use the x11 and wayland for displaying the app.
I use IPC because I need gnome-terminal to open the ssh terminal
And Talk because without it, the gnome shell doesn't spawn.

### CODE REVIEW

**Comment by SStoklosa23** [comment]
bot, build io.github.gssh

**Comment by SStoklosa23** [comment]
bot, build io.github.gssh-linux.gssh

File: io.github.gssh-linux.gssh.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+app-id: io.github.gssh-linux.gssh
```
Comment (hfiguiere) [inline io.github.gssh-linux.gssh.yml:1]:
```suggestion
app-id: io.github.gssh_linux.gssh
```

Comment (hfiguiere) [reply io.github.gssh-linux.gssh.yml:1]: source: https://docs.flatpak.org/en/latest/conventions.html#application-ids

Comment (SStoklosa23) [reply io.github.gssh-linux.gssh.yml:1]: I can't do that, the name of the org is gssh-linux

Comment (hfiguiere) [reply io.github.gssh-linux.gssh.yml:1]: you didn't read the doc through. It says `-` is replaced by `_` except in the last component.

Sadly I need to fix the linter to check for this.

**Comment by SStoklosa23** [comment]
bot, build io.github.gssh-linux.gssh

**Comment by SStoklosa23** [comment]
bot, build io.github.gssh_linux.gssh

**Comment by SStoklosa23** [comment]
bot, build io.github.gssh_linux.gssh.yml

**Comment by SStoklosa23** [comment]
bot, build io.github.gssh_linux.gssh

**Comment by SStoklosa23** [comment]
bot, build io.github.gssh_linux.gssh

**Comment by SStoklosa23** [comment]
bot, build io.github.gssh_linux.gssh

**Comment by SStoklosa23** [comment]
bot, build io.github.gssh_linux.gssh

File: io.github.gssh_linux.gssh.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  - name: vte
+    buildsystem: meson
+    config-opts:
+      - "-Dglade=false"
+      - "-Dgtk3=true"
+      - "-Dgtk4=false"
+      - "-Dvapi=false"
+    sources:
+      - type: git
+        url: https://gitlab.gnome.org/GNOME/vte.git
```
Comment (hfiguiere) [inline io.github.gssh_linux.gssh.yml:35]: usually for GNOME packages we use tarballs.

Comment (hfiguiere) [reply io.github.gssh_linux.gssh.yml:35]: see https://download.gnome.org/sources/vte/0.74/

File: io.github.gssh_linux.gssh.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        tag: vte-0-74
+
+  - name: org.gnome.Platform
+    version: 44
+    runtime: org.gnome.Platform
+    sdk: org.gnome.Sdk
+    override: true
+    build-options:
+      # Add the Vte extension
+      - --share=lib/vte
```
Comment (hfiguiere) [inline io.github.gssh_linux.gssh.yml:45]: not sure what you are doing here but this is invalid.

```
flatpak-builder:2): flatpak-builder-WARNING **: 00:49:15.365: Unknown property runtime-options for type BuilderManifest
(flatpak-builder:2): flatpak-builder-WARNING **: 00:49:15.365: Unknown property version for type BuilderModule
(flatpak-builder:2): flatpak-builder-WARNING **: 00:49:15.365: Unknown property runtime for type BuilderModule
(flatpak-builder:2): flatpak-builder-WARNING **: 00:49:15.365: Unknown property sdk for type BuilderModule
(flatpak-builder:2): flatpak-builder-WARNING **: 00:49:15.365: Unknown property override for type BuilderModule
(flatpak-builder:2): Json-WARNING **: 00:49:15.365: Failed to deserialize "build-options" property of type "BuilderOptions" for an object of type "BuilderModule"
```

File: io.github.gssh_linux.gssh.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+runtime: org.gnome.Platform
+runtime-version: '44'
+runtime-options:
+sdk: org.gnome.Sdk
+command: runner.sh
+finish-args:
+  - --share=network
+  - --socket=x11
+  - --share=ipc
+  - --env=GI_TYPELIB_PATH=/app/lib/girepository-1.0
```
Comment (bbhtt) [inline io.github.gssh_linux.gssh.yml:11]: This should be set inside the sandbox:

```
>>> GIRepository.Repository.get_search_path()
['/app/lib/girepository-1.0', '/usr/lib/x86_64-linux-gnu/girepository-1.0']
```

File: io.github.gssh_linux.gssh.yml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+modules:
+  - name: Gssh
+    buildsystem: simple
+    build-commands:
+      - install -D Flatpak/runner.sh /app/bin/runner.sh
+      - install -D main.py /app/bin/main.py
+      - install -D Flatpak/.desktop /app/share/applications/io.github.gssh_linux.gssh.desktop
+      - install -D Flatpak/appicon.png /app/share/icons/hicolor/128x128/apps/io.github.gssh_linux.gssh.appicon.png
```
Comment (bbhtt) [inline io.github.gssh_linux.gssh.yml:19]: Icon name doesn't match the app-id.

File: io.github.gssh_linux.gssh.yml (hunk 4 above)
Comment (bbhtt) [reply io.github.gssh_linux.gssh.yml:11]: https://github.com/GNOME/gnome-build-meta/blob/1c4e2a340a62576f2ffc7b8f2ea9a589a9269036/elements/flatpak/platform.bst#L24

File: io.github.gssh_linux.gssh.yml (hunk 5 above)
Comment (TingPing) [reply io.github.gssh_linux.gssh.yml:19]: It is valid but a bit unusual naming.

Comment (TingPing) [reply io.github.gssh_linux.gssh.yml:19]: Also general recommendation, these files have *nothing* to do with flatpak. They are files that will always be installed for every package. So it is strange to just shove them in a `Flatpak` directory.

Comment (hfiguiere) [reply io.github.gssh_linux.gssh.yml:19]: it'll get exported and the desktop file reference it like that. But I'm also of the opinion that this is a bad practice. And I'm not sure if it doesn't have any obscure side effect down the line.

Also I recommend naming the desktop file properly in the upstream repository. `.desktop` is a bad idea as it starts with a `.`.

Comment (hfiguiere) [reply io.github.gssh_linux.gssh.yml:19]: also that image is 512x512 so its install location is incorrect.

File: io.github.gssh_linux.gssh.yml [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  - --share=network
+  - --socket=x11
+  - --share=ipc
+  - --env=GI_TYPELIB_PATH=/app/lib/girepository-1.0
+modules:
+  - name: Gssh
+    buildsystem: simple
+    build-commands:
+      - install -D Flatpak/runner.sh /app/bin/runner.sh
+      - install -D main.py /app/bin/main.py
```
Comment (hfiguiere) [inline io.github.gssh_linux.gssh.yml:17]:
```suggestion
      - install -Dm755 Flatpak/runner.sh /app/bin/runner.sh
      - install -Dm755 main.py /app/bin/main.py
```

It's safer to make sure they have +x mode set on install.

File: io.github.gssh_linux.gssh.yml [hunk 7]
```diff
@@ -0,0 +1,45 @@
+app-id: io.github.gssh_linux.gssh
+runtime: org.gnome.Platform
+runtime-version: '44'
+runtime-options:
```
Comment (hfiguiere) [inline io.github.gssh_linux.gssh.yml:4]:
```suggestion
```

File: io.github.gssh_linux.gssh.yml [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+  - --share=network
+  - --socket=x11
+  - --share=ipc
+  - --env=GI_TYPELIB_PATH=/app/lib/girepository-1.0
+modules:
+  - name: Gssh
+    buildsystem: simple
+    build-commands:
+      - install -Dm755 Flatpak/runner.sh /app/bin/runner.sh
+      - install -Dm755 main.py /app/bin/main.py
```
Comment (TingPing) [inline io.github.gssh_linux.gssh.yml:16]:
```suggestion
      - install -D main.py /app/bin/main.py
```

This isn't executable.

File: io.github.gssh_linux.gssh.yml [hunk 9]
```diff
@@ -0,0 +1,44 @@
+app-id: io.github.gssh_linux.gssh
+runtime: org.gnome.Platform
+runtime-version: '44'
+sdk: org.gnome.Sdk
+command: runner.sh
+finish-args:
+  - --share=network
+  - --socket=x11
```
Comment (TingPing) [inline io.github.gssh_linux.gssh.yml:8]:
```suggestion
  - --socket=fallback-x11
  - --socket=wayland
```

File: io.github.gssh_linux.gssh.yml (hunk 3 above)
Comment (SStoklosa23) [reply io.github.gssh_linux.gssh.yml]: Useless so I removed it.

**Comment by SStoklosa23** [comment]
bot, build io.github.gssh_linux.gssh

**Comment by SStoklosa23** [comment]
bot, build io.github.gssh_linux.gssh

File: io.github.gssh_linux.gssh.yml (hunk 4 above)
Comment (SStoklosa23) [reply io.github.gssh_linux.gssh.yml]: Yes, it was not needed so I removed it.

File: io.github.gssh_linux.gssh.yml [hunk 10]
```diff
@@ -0,0 +1,38 @@
+app-id: io.github.gssh_linux.gssh
+runtime: org.gnome.Platform
+runtime-version: '44'
```
Comment (hfiguiere) [inline io.github.gssh_linux.gssh.yml:3]: 45 is out now.

**Pull request closed without merging**